src/Microsoft.ML.Data/Transforms/ColumnBindingsBase.cs

```diff
@@ -14,16 +14,40 @@
 
 namespace Microsoft.ML.Data
 {
-    internal abstract class SourceNameColumnBase
+    public class OneToOneColumn
     {
+        /// <summary>Name of the column resulting from the transformation of <see cref="InputColumnName"/>.</summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Name of the new column", ShortName = "name")]
-        public string Name;
+        public string OutputColumnName;
+
+        /// <summary>Name of the column to transform. If set to <see langword="null"/>, the value of the <see cref= "OutputColumnName"/> will be used as source.</summary>
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Name of the source column", ShortName = "source, src")]
```

Comment:
>ShortName [](start = 83, length = 9)

Note that you could have used the `Name` field here. That might have been preferable if we care about not changing the behavior of command line and entry-points, at least not yet. (We can always adjust later if we really want to.)

Reply:
Ok will use Name = .... to prevent further changes

---
In reply to: [267542069](https://github.com/dotnet/machinelearning/pull/3033#discussion_r267542069) [](ancestors = 267542069)